compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/QueryDesugar.java

```diff
@@ -1955,10 +1984,23 @@ public void visit(BLangSimpleVarRef bLangSimpleVarRef) {
                     }
                 }
                 identifiers.put(identifier, symbol);
+                newIdentifiers.put(identifier, symbol);
             } else if (identifiers.containsKey(identifier) && (withinLambdaOrArrowFunc || withinQuery)) {
                 symbol = identifiers.get(identifier);
+                if (symbol.closure) {
+                    SymbolEnv queryLambdaEnv = SymbolEnv.createFuncBodyEnv(currentQueryLambdaBody, env);
+                    queryLambdaEnv.scope.define(symbol.name, symbol);
+                }
                 bLangSimpleVarRef.symbol = symbol;
                 bLangSimpleVarRef.varSymbol = symbol;
+            } else if (!symbol.closure && newIdentifiers.containsKey(identifier)) {
```

Comment:
Why do we need to perform the check `newIdentifiers.containsKey(identifier)`?

Reply:
Lets say we are in the do clause body. I need to identify whether that identifier coming from a previous clause or its in coming from the doClause.